### PR TITLE
chore: remove unused AnticheatSetSkipOnePacketForASH() function

### PR DESCRIPTION
**Pull request by natrist**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Remove unused AnticheatSetSkipOnePacketForASH() function from what seems to be remnants of an anti-cheat system that is no longer present in code

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Handlers/TaxiHandler.cpp [hunk 1]
```diff
@@ -107,8 +107,7 @@ void WorldSession::SendTaxiMenu(Creature* unit)
     GetPlayer()->SetTaxiCheater(lastTaxiCheaterState);
 }
 
-void WorldSession::SendDoFlight(uint32 mountDisplayId, uint32 path, uint32 pathNode)
-{
+void WorldSession::SendDoFlight(uint32 mountDisplayId, uint32 path, uint32 pathNode) {
```
Comment (Winfidonarleyan) [inline src/server/game/Handlers/TaxiHandler.cpp:110]: ?

Comment (natrist) [reply src/server/game/Handlers/TaxiHandler.cpp:110]: I don't remember changing that line. 😂
Must be the IDE but either way I've changed it back to your preference.

File: src/server/game/Handlers/TaxiHandler.cpp [hunk 2]
```diff
@@ -119,14 +119,10 @@ void WorldSession::SendDoFlight(uint32 mountDisplayId, uint32 path, uint32 pathN
     if (mountDisplayId)
         GetPlayer()->Mount(mountDisplayId);
 
-    if (Creature* critter = ObjectAccessor::GetCreature(*GetPlayer(), GetPlayer()->GetCritterGUID()))
-    {
+    if (Creature *critter = ObjectAccessor::GetCreature(*GetPlayer(), GetPlayer()->GetCritterGUID()))
         critter->DespawnOrUnsummon();
-    }
```
Comment (Kitzunu) [inline src/server/game/Handlers/TaxiHandler.cpp:125]: ?

File: src/server/game/Handlers/TaxiHandler.cpp [hunk 3]
```diff
@@ -119,14 +119,10 @@ void WorldSession::SendDoFlight(uint32 mountDisplayId, uint32 path, uint32 pathN
     if (mountDisplayId)
         GetPlayer()->Mount(mountDisplayId);
 
-    if (Creature* critter = ObjectAccessor::GetCreature(*GetPlayer(), GetPlayer()->GetCritterGUID()))
-    {
+    if (Creature *critter = ObjectAccessor::GetCreature(*GetPlayer(), GetPlayer()->GetCritterGUID()))
```
Comment (Kitzunu) [inline src/server/game/Handlers/TaxiHandler.cpp:122]:
```suggestion
    if (Creature* critter = ObjectAccessor::GetCreature(*GetPlayer(), GetPlayer()->GetCritterGUID()))
```

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**